packages/chakra-ui/src/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { IconButton } from "@chakra-ui/react";
+import { AddIcon, MinusIcon, ArrowUpIcon, ArrowDownIcon } from "@chakra-ui/icons";
+
+const mappings = {
+  "remove": <MinusIcon />,
+  "plus": <AddIcon />,
+  "arrow-up": <ArrowUpIcon />,
+  "arrow-down": <ArrowDownIcon />,
+};
+
+const ChakraIconButton = (props: { [x: string]: any; iconMap: any; }) => {
+  const { iconMap, ...otherProps } = props;
+  // @ts-ignore
+  return <IconButton {...otherProps} icon={mappings[iconMap] ? mappings[iconMap] : <AddIcon />} />;
+};
```

Comment:
What do you think of this?

```suggestion
type IconButtonProps = Omit<ChakraIconButtonProps, "aria-label" | "icon"> & {
  icon: keyof typeof mappings;
};

const ChakraIconButton = (props: IconButtonProps) => {
  const { icon, ...otherProps } = props;
  return <IconButton {...otherProps}  icon={mappings[icon]} aria-label={icon}/>;
};
```
This forces an `icon` prop to be used, if we need to make it optional we can then default it to `"plus"`
We can then make sure we are using `icon` instead of `iconMap` where it's used.